src/main/java/fi/aalto/cs/apluscourses/utils/content/ZippedDir.java

```diff
@@ -0,0 +1,24 @@
+package fi.aalto.cs.apluscourses.utils.content;
+
+import fi.aalto.cs.apluscourses.utils.DirAwareZipFile;
+import fi.aalto.cs.apluscourses.utils.RemoteFileCache;
+import java.io.IOException;
+import java.nio.file.Path;
+import org.jetbrains.annotations.NotNull;
+
+public class ZippedDir implements Content {
```

Comment:
The name of this and `ZippedFile` are potentially a bit confusing, since they are actually used to copy remote files to local files

Reply:
You're right. I wanted `Content` interface to be generic enough that the content can be from any source (local, remote, resource...). This class's name would better be `RemoteZippedDir` or something, I guess.